chips/stm32f4xx/src/clocks/pll.rs

```diff
@@ -202,6 +206,29 @@ impl<'a, PllConstants: clock_constants::PllConstants> Pll<'a, PllConstants> {
         unreachable!("The previous for loop should always return");
     }
 
+    /// Get the PLL source clock frequency in MHz
+    ///
+    /// # Returns
+    ///
+    /// + [HSI_FREQUENCY_MHZ]: if the pll source is HSI
+    /// + HSE frequency if the pll source is HSE
+    fn get_pll_source_frequency(&self) -> usize {
+        if self.rcc.is_hsi_clock_system_clock() {
+            HSI_FREQUENCY_MHZ
+        } else {
+            self.rcc.get_hse_frequency().unwrap()
```

Comment:
When I wrote this function, I didn't think of the case when the source can be disabled. HSI could never be disabled because it was either used as a direct system clock source or through PLL. Now, it could be disabled in the case where HSE is used as a system clock. I suggest changing the return type of this function to `Option<usize>` and remove the call to `unwrap` that can cause a panic.

Reply:
I changed it a bit, but the functionality is the same. I think `unwrap` is OK since it does not make sense if the HSE is the system clock but it's frequency is not configured

Reply:
See [this comment](https://github.com/tock/tock/pull/3750#discussion_r1407796878)